src/persist-client/src/write.rs

```diff
@@ -457,6 +594,39 @@ where
     }
 }
 
+/// A handle to a batch of updates that has been written to blob storage but
+/// which has not yet been appended to a shard.
+// TODO(aljoscha): It seems we would need an async drop impl that cleans up the
```

Comment:
yeah best we can do is some sort of manual cleanup async call and a Drop impl that logs when they haven't been cleaned up. probably worth fixing this while we're in here instead of adding the TODO